sdk/storage/azure-storage-common/inc/azure/storage/common/internal/storage_bearer_token_authentication_policy.hpp

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <azure/core/http/policies/policy.hpp>
+
+#include <mutex>
+#include <shared_mutex>
+
+namespace Azure { namespace Storage { namespace _internal {
+
+  class StorageBearerTokenAuthenticationPolicy final
+      : public Core::Http::Policies::_internal::BearerTokenAuthenticationPolicy {
+  public:
+    /**
+     * @brief Construct a Storage Bearer Token challenge authentication policy.
+     *
+     * @param credential An #Azure::Core::TokenCredential to use with this policy.
+     * @param tokenRequestContext A context to get the token in.
+     * @param enableTenantDiscovery Enables tenant discovery through the authorization challenge.
+     */
+    explicit StorageBearerTokenAuthenticationPolicy(
+        std::shared_ptr<const Azure::Core::Credentials::TokenCredential> credential,
+        Azure::Core::Credentials::TokenRequestContext tokenRequestContext,
+        bool enableTenantDiscovery)
+        : BearerTokenAuthenticationPolicy(std::move(credential), tokenRequestContext),
+          m_scopes(tokenRequestContext.Scopes), m_safeTenantId(tokenRequestContext.TenantId),
+          m_enableTenantDiscovery(enableTenantDiscovery)
+    {
+    }
+
+    ~StorageBearerTokenAuthenticationPolicy() override {}
+
+    std::unique_ptr<HttpPolicy> Clone() const override
+    {
+      return std::unique_ptr<HttpPolicy>(new StorageBearerTokenAuthenticationPolicy(*this));
+    }
+
+  private:
+    struct SafeTenantId
+    {
+    public:
+      explicit SafeTenantId(std::string tenantId) : m_tenantId(std::move(tenantId)) {}
+
+      SafeTenantId(const SafeTenantId& other) : m_tenantId(other.Get()) {}
+      SafeTenantId& operator=(const SafeTenantId& other)
```

Comment:
Is `SafeTenantId` or `StorageBearerTokenAuthenticationPolicy` required to be copy-assignable?

Reply:
No, I think clone() method is enough.